src/seedu/addressbook/commands/DeleteCommand.java

```diff
@@ -37,13 +41,22 @@ public CommandResult execute() {
             final ReadOnlyPerson target = getTargetPerson();
             privilege.checkTargetIsSelf(target);
             addressBook.removePerson(target);
+            Set<Exam> examSet = target.getExams();
+            for (Exam e: examSet) {
+                Exam examAfter = new Exam(e);
+                examAfter.setTakers(examAfter.getTakers() - 1);
+                examBook.updateExam(e, examAfter);
+                addressBook.updateExam(e, examAfter);
+            }
```

Comment:
This seems pretty inefficient in terms of object usage. You are creating a new object for every exam in the set rather than just modifying the target object itself

Reply:
because its a set of exams. so remove and add?

Reply:
Since you have reference to the original object(e), can't you just modify it instead rather than instantiating a new object using e as a parameter?

Reply:
when deleting a person, there's also printing out of the full details of the person(since you want to know what you have deleted). if the exam is edited, it will not print the original parameter.

Reply:
I also hashed the number of takers for exams as well, to keep the check for deleting correctly. In Exam when I override the equal however, i did not include check for takers in, this is because the method that calls for equals is in adding an exam. I feel that number of takers is not necessary to check as the base details of the exams are the others.